src/core/getProfits.ts

```diff
@@ -35,33 +44,47 @@ export async function getProfits({
           blockNumber: toBlock,
           protocolTokenAddresses,
         })
-        .then(aggregateFiatBalances),
+        .then((result) => {
+          rawEndPositionValues = result
+          return aggregateFiatBalances(result)
+        }),
       adapter
         .getPositions({
           userAddress,
           blockNumber: fromBlock,
           protocolTokenAddresses,
         })
-        .then(aggregateFiatBalances),
+        .then((result) => {
+          rawStartPositionValues = result
+          return aggregateFiatBalances(result)
+        }),
     ])
   } else {
     // Call toBlock first and filter fromBlock
-    endPositionValues = await adapter
+    ;(endPositionValues = await adapter
```

Comment:
Is this intended? I mean the `;(` part.

In this branch, we want to first run the adapter with the end positions, and then pass those positions as a filter.